settings.py

```diff
@@ -148,26 +86,12 @@ def create_options_dict(options: Sequence[BaseOption]) -> dict[str, JSON]:
     """
     settings: dict[str, JSON] = {}
     for option in options:
-        match option:
-            case ValueOption():
-                # The generics mean the type of value is technically unknown here
-                value = cast(JSON, option.value)  # type: ignore
-                settings[option.identifier] = value
-
-            case GroupedOption() | NestedOption():
-                settings[option.identifier] = create_options_dict(option.children)
-
-            # Button option is the only standard option which is not abstract, but also not a value,
-            # and doesn't have children.
-            # Just no-op it so that it doesn't show an error
-            case ButtonOption():
-                pass
-
-            case _:
-                logging.error(
-                    f"Couldn't save settings for unknown option type {type(option).__name__}",
-                )
-
+        # Button option is the only standard option which is not abstract, but also not a value,
+        # and doesn't have children.
+        # Just no-op it so that it doesn't show an error
+        if isinstance(option, ButtonOption):
+            continue
+        settings[option.identifier] = option._to_json()  # type: ignore
```

Comment:
should be able to turn this into a dict comprehension once the issue with button is solved